### ISSUE TITLE
Add a LICENSE file (and refer to it in the README)

### ISSUE THREAD

**Issue by Tyil** [vim-perl/vim-perl6#18]
# Description of issue
This project currently does not carry a LICENSE. In the spirit of free software I'd like it to adopt a free license clearly in the project root. If possible, refer to the license in the README.md as well.

**Comment by hoelzro** [comment]
@Tyil this is a great idea - I wonder if we technically need contributors' permission first? If not, would you mind drafting a PR to add this to the repo?

**Comment by Tyil** [comment]
I would've drafter a PR already, but I'm not sure what license it would fall under. I'm personally a big fan of (A)GPL, but I'm not sure what you or other contributors would think about that.

Since the project is currently unlicensed, that mean everyone has to agree on the license picked as well if I'm correct. But I'm not a lawyer, so don't take it as legal advice. I'd like input from all contributors anyway, since it is a community project after all.

**Comment by hoelzro** [comment]
I think the typical choice for Vimscript is the Vim license itself - I think that you may be right about the "everyone must agree" part =/

**Comment by Tyil** [comment]
It's 15 contributors. What would be the best way to contact them? Just send everyone an email and hope they respond? We can tag them each in the issue as well, perhaps they'll get a notification about that too.

**Comment by hoelzro** [comment]
I guess let's try tagging them in this issue - otherwise, trying e-mail would be a good idea.

**Comment by Tyil** [comment]
ping @0racle, @atweiden, @blueyed, @cfa, @eiro, @grondilu, @hinrik, @itchyny, @KrisShannon, @petdance, @samcv, @simmel, @Util

Please respond with your LICENSE options for this codebase.

**Comment by simmel** [comment]
I'd prefer a permissive license like `ISC`, `MIT` or `BSD` but I can cave in if everyone else wants something different.

**Comment by blueyed** [comment]
I am fine with any open license - have not checked, but my contribution is probably very small anyway.

**Comment by itchyny** [comment]
Good idea to add LICENSE. Personally I'm a fan of MIT but I'll follow the decision of main contributors. My change to this repo is just a nitpick.

**Comment by petdance** [comment]
I can't imagine why we wouldn't license it with whatever vim's license is.

**Comment by petdance** [comment]
Seems to me that we should use whatever license Vim uses, especially if this gets folded into the main vim distribution.

**Comment by cfa** [comment]
I'm happy with anything—my contribution was minor.

**Comment by hoelzro** [comment]
FWIW, choosing something _other_ than the Vim license might limit our ability to have changes to this incorporated with the upstream Vim project.

**Comment by petdance** [comment]
There would have to be a pretty strong argument to NOT use Vim's license.

**Comment by grondilu** [comment]
I'm ok with MIT, GPL or whatever Vim uses,

**Comment by Tyil** [comment]
First half of the contributors have responded, thanks for your feedback thus far people!

**Comment by samcv** [comment]
I'm ok with any any OSI approved software license.
